rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/packagemapping/PackageNameMapping.java

```diff
@@ -68,6 +75,76 @@ public Rule toOrganization(String organization)
         return this;
     }
 
+    public static boolean isExclusivelyKnownArchive(GraphRewrite event, String filePath)
+    {
+        String extension = StringUtils.substringAfterLast(filePath, ".");
+
+        if (!StringUtils.equalsIgnoreCase(extension, "jar"))
+            return false;
+
+        ZipFile archive;
+        try
+        {
+            archive = new ZipFile(filePath);
+        } catch (IOException e)
+        {
+            return false;
+        }
+
+        WindupJavaConfigurationService javaConfigurationService = new WindupJavaConfigurationService(event.getGraphContext());
+        WindupJavaConfigurationModel javaConfigurationModel = WindupJavaConfigurationService.getJavaConfigurationModel(event.getGraphContext());
+
+        // indicates that the user did specify some packages to scan explicitly (as opposed to scanning everything)
+        boolean customerPackagesSpecified = javaConfigurationModel.getScanJavaPackages().iterator().hasNext();
+
+        // this should only be true if:
+        // 1) the package does not contain *any* customer packages.
+        // 2) the package contains "known" vendor packages.
+        boolean exclusivelyKnown = false;
+
+        String organization = null;
+        Enumeration<?> e = archive.entries();
+
+        // go through all entries...
+        ZipEntry entry;
+        while (e.hasMoreElements())
+        {
+            entry = (ZipEntry) e.nextElement();
+            String entryName = entry.getName();
+
```

Comment:
For the record, We've discussed that on video chat. I need a bit different kind of info about packages.